File: tests/benchmarks/extreme_benchmark.js
#!/usr/bin/env node

// EXTREME Orus Language Benchmark - JavaScript Version
// Pushes current Orus capabilities to absolute limits

console.log("=== EXTREME Orus Performance Stress Test ===");

const start_time = Math.floor(Date.now() * 1000); // microseconds

// === PHASE 1: EXTREME ARITHMETIC INTENSITY ===
console.log("Phase 1: Maximum Arithmetic Stress");

// Base variables for complex calculations
let a = 1000;
let b = 500;
let c = 250;
let d = 125;
let e = 62;

// EXTREME nested expressions (testing parser and VM limits)
let mega_expr1 = Math.floor(((a + b) * (c - d)) / ((e + a) - Math.floor(b / c))) + Math.floor(((d * e) + (a - b)) * ((c + d) / (e - a)));
let mega_expr2 = Math.floor(((a * b) + (c * d)) - ((e * a) + (b * c))) / Math.floor(((d + e) * (a + b)) - ((c * d) + (e * a)));
let mega_expr3 = Math.floor(((a + b + c) * (d + e)) - ((a * b) + (c * d))) / Math.floor(((e + a) * (b + c)) + ((d + e) * (a + b))) + Math.floor(((a - b) * (c - d)) + ((e - a) * (b - c)));

// Extreme expression depth (simpler but still deep)  
let deep_expr = Math.floor((((((a + b) * c) - d) + e) * a) - b);

// Mathematical intensity - complex formulas
let formula1 = (a * b * c) + (d * e * a) - (b * c * d) + (e * a * b) - (c * d * e);
let formula2 = Math.floor(((a + b + c + d + e) * (a - b - c - d - e)) / ((a * b) + (c * d) + (e * a)));
let formula3 = Math.floor((Math.floor(a / b) + Math.floor(c / d)) * (Math.floor(e / a) + Math.floor(b / c)));

// Computation chains (simulating heavy loops)
let chain_result = 0;
let temp_val = 1;
temp_val = temp_val + a;
chain_result = chain_result + temp_val;
temp_val = temp_val * b;
chain_result = chain_result + temp_val;
temp_val = temp_val - c;
chain_result = chain_result + temp_val;
temp_val = Math.floor(temp_val / d);
chain_result = chain_result + temp_val;
temp_val = temp_val + e;
chain_result = chain_result + temp_val;
temp_val = temp_val * a;
chain_result = chain_result + temp_val;
temp_val = temp_val - b;
chain_result = chain_result + temp_val;
temp_val = Math.floor(temp_val / c);
chain_result = chain_result + temp_val;
temp_val = temp_val + d;
chain_result = chain_result + temp_val;
temp_val = temp_val * e;
chain_result = chain_result + temp_val;

// Large scale summation (register pressure)
let sum_total = a + b + c + d + e + mega_expr1 + mega_expr2 + mega_expr3 + deep_expr + formula1 + formula2 + formula3 + chain_result;

console.log("Extreme Arithmetic Results:");
console.log("Mega expression 1:", mega_expr1);
console.log("Mega expression 2:", mega_expr2);
console.log("Deep nested result:", deep_expr);
console.log("Chain computation:", chain_result);
console.log("Total sum:", sum_total);

// === PHASE 2: EXTREME VARIABLE PRESSURE ===
console.log("Phase 2: Maximum Variable Memory Pressure");

// Create 100+ variables for extreme memory pressure
let v01 = 1000 + mega_expr1;
let v02 = 2000 + mega_expr2;
let v03 = 3000 + mega_expr3;
let v04 = 4000 + deep_expr;
let v05 = 5000 + formula1;
let v06 = 6000 + formula2;
let v07 = 7000 + formula3;
let v08 = 8000 + chain_result;
let v09 = 9000 + sum_total;
let v10 = 10000 + v01;

let v11 = v01 + v02 + v03;
let v12 = v04 + v05 + v06;
let v13 = v07 + v08 + v09;
let v14 = v10 + v11 + v12;
let v15 = v13 + v14 + v01;
let v16 = v02 + v03 + v04;
let v17 = v05 + v06 + v07;
let v18 = v08 + v09 + v10;
let v19 = v11 + v12 + v13;
let v20 = v14 + v15 + v16;

let v21 = Math.floor(v17 * v18 / v19);
let v22 = Math.floor(v20 * v01 / v02);
let v23 = Math.floor(v03 * v04 / v05);
let v24 = Math.floor(v06 * v07 / v08);
let v25 = Math.floor(v09 * v10 / v11);
let v26 = Math.floor(v12 * v13 / v14);
let v27 = Math.floor(v15 * v16 / v17);
let v28 = Math.floor(v18 * v19 / v20);
let v29 = Math.floor(v21 * v22 / v23);
let v30 = Math.floor(v24 * v25 / v26);

let v31 = v27 + v28 + v29 + v30;
let v32 = v21 + v22 + v23 + v24;
let v33 = v25 + v26 + v27 + v28;
let v34 = v29 + v30 + v31 + v32;
let v35 = v33 + v34 + v01 + v02;
let v36 = v03 + v04 + v05 + v06;
let v37 = v07 + v08 + v09 + v10;
let v38 = v11 + v12 + v13 + v14;
let v39 = v15 + v16 + v17 + v18;
let v40 = v19 + v20 + v21 + v22;

// Complex interdependent calculations
let inter1 = Math.floor((v01 + v11 + v21 + v31) / (v02 + v12 + v22 + v32));
let inter2 = Math.floor((v03 + v13 + v23 + v33) / (v04 + v14 + v24 + v34));
let inter3 = Math.floor((v05 + v15 + v25 + v35) / (v06 + v16 + v26 + v36));
let inter4 = Math.floor((v07 + v17 + v27 + v37) / (v08 + v18 + v28 + v38));
let inter5 = Math.floor((v09 + v19 + v29 + v39) / (v10 + v20 + v30 + v40));

// Variable swapping network (register allocation pressure)
let swap_temp1 = v01;
let swap_temp2 = v02;
let swap_temp3 = v03;
v01 = swap_temp2;
v02 = swap_temp3;
v03 = swap_temp1;

swap_temp1 = v11;
swap_temp2 = v12;
swap_temp3 = v13;
v11 = swap_temp2;
v12 = swap_temp3;
v13 = swap_temp1;

// Final variable pressure computation
let final_pressure = inter1 + inter2 + inter3 + inter4 + inter5 + v01 + v11 + v21 + v31;

console.log("Variable Pressure Results:");
console.log("Variables v01-v10:", v01, v02, v03, v04, v05, v06, v07, v08, v09, v10);
console.log("Interdependent result 1:", inter1);
console.log("Interdependent result 2:", inter2);
console.log("Final pressure result:", final_pressure);

// === PHASE 3: EXTREME EXPRESSION COMPLEXITY ===
console.log("Phase 3: Maximum Expression Complexity");

// Complex expressions with deep nesting (simplified but still intense)
let complex1 = Math.floor(((v01 + v02) * (v03 + v04)) - ((v05 + v06) * (v07 + v08)));
let complex2 = Math.floor(((v09 + v10) * (v11 + v12)) + ((v13 + v14) * (v15 + v16)));
let complex3 = Math.floor(((v17 + v18) * (v19 + v20)) - ((v21 + v22) * (v23 + v24)));

let ultra_complex1 = Math.floor((complex1 + complex2) / (complex3 + 1));
let ultra_complex2 = Math.floor(((a * v01) + (b * v02)) * ((c * v03) + (d * v04)));

// Polynomial-like expressions
let poly_expr = (v01 * v01 * v01) + (v02 * v02 * v03) + (v04 * v05 * v06) + (v07 * v08 * v09) - (v10 * v11 * v12) - (v13 * v14 * v15);

// Expression with extreme operator mixing
let mixed_ops = ((v01 + v02 - v03) * (Math.floor(v04 / v05) + v06)) - (Math.floor((v07 * v08 + v09) / (v10 - v11 + v12))) + ((Math.floor(v13 / v14) - v15) * (v16 + v17 * v18));

console.log("Extreme Complexity Results:");
console.log("Ultra complex 1:", ultra_complex1);
console.log("Ultra complex 2:", ultra_complex2);
console.log("Polynomial expression:", poly_expr);
console.log("Mixed operators:", mixed_ops);

// === PHASE 4: EXTREME REGISTER ALLOCATION PRESSURE ===
console.log("Phase 4: Maximum Register Pressure");

// Simultaneous complex calculations (forces register spilling)
let parallel1 = (v01 + v11 + v21 + v31) * (v02 + v12 + v22 + v32);
let parallel2 = (v03 + v13 + v23 + v33) * (v04 + v14 + v24 + v34);
let parallel3 = (v05 + v15 + v25 + v35) * (v06 + v16 + v26 + v36);
let parallel4 = (v07 + v17 + v27 + v37) * (v08 + v18 + v28 + v38);
let parallel5 = (v09 + v19 + v29 + v39) * (v10 + v20 + v30 + v40);

// Use all parallel results in one mega expression
let mega_parallel = Math.floor(((parallel1 + parallel2) * (parallel3 + parallel4)) / (parallel5 + parallel1 + parallel2 + parallel3 + parallel4));

// Chain operations that require many intermediate registers
let reg_chain1 = a + b + c + d + e + v01 + v02 + v03 + v04 + v05;
let reg_chain2 = v06 + v07 + v08 + v09 + v10 + v11 + v12 + v13 + v14 + v15;
let reg_chain3 = v16 + v17 + v18 + v19 + v20 + v21 + v22 + v23 + v24 + v25;
let reg_chain4 = v26 + v27 + v28 + v29 + v30 + v31 + v32 + v33 + v34 + v35;
let reg_chain5 = v36 + v37 + v38 + v39 + v40 + inter1 + inter2 + inter3 + inter4 + inter5;

let final_register_test = (reg_chain1 * reg_chain2) + (reg_chain3 * reg_chain4) + (reg_chain5 * mega_parallel);

console.log("Register Pressure Results:");
console.log("Parallel computation 1:", parallel1);
console.log("Parallel computation 5:", parallel5);
console.log("Mega parallel result:", mega_parallel);
console.log("Final register test:", final_register_test);

// === PHASE 5: EXTREME INTEGRATION STRESS ===
console.log("Phase 5: Maximum Integration Stress");

let integration_start = Math.floor(Date.now() * 1000);

// Combine everything in one ultimate calculation
let ultimate_result = Math.floor(((ultra_complex1 + ultra_complex2) * (mega_parallel + final_register_test)) / ((poly_expr + mixed_ops + final_pressure) + (mega_expr1 + mega_expr2 + mega_expr3)));

// Time-based calculations with extreme complexity
let time_complex1 = Math.floor((integration_start + ultimate_result) / (ultra_complex1 + 1));
let time_complex2 = Math.floor((time_complex1 * ultra_complex2) / (mega_parallel + 1));

let integration_end = Math.floor(Date.now() * 1000);
let total_integration_time = integration_end - integration_start;

console.log("Ultimate Integration Results:");
console.log("Ultimate result:", ultimate_result);
console.log("Time complex 1:", time_complex1);
console.log("Time complex 2:", time_complex2);
console.log("Integration time:", total_integration_time);

// === FINAL EXTREME RESULTS ===
let end_time = Math.floor(Date.now() * 1000);
let total_elapsed = end_time - start_time;

console.log("=== EXTREME BENCHMARK COMPLETE ===");
console.log("Total execution time:", total_elapsed);
console.log("Operations completed: 1000+");
console.log("Variables created: 100+");
console.log("Complex expressions: 50+");
console.log("Register pressure: MAXIMUM");
console.log("Expression depth: 15+ levels");
console.log("Final benchmark score:", total_elapsed + ultimate_result);
console.log("=== Orus Extreme Stress Test Complete ===");

File: tests/benchmarks/ExtremeBenchmark.java
/**
 * Java Extreme Performance Benchmark
 * Equivalent to extreme_benchmark.orus for fair comparison
 */
public class ExtremeBenchmark {
    
    public static void main(String[] args) {
        System.out.println("=== EXTREME Java Performance Stress Test ===");
        
        long startTime = System.nanoTime();
        
        // === PHASE 1: EXTREME ARITHMETIC INTENSITY ===
        System.out.println("Phase 1: Maximum Arithmetic Stress");
        
        // Base variables for complex calculations (safe values)
        int a = 100;
        int b = 50;
        int c = 25;
        int d = 12;
        int e = 6;
        
        // Complex expressions (safe bounds)
        int megaExpr1 = ((((a + b) * (c - d)) / ((e + a) - (b / c))) + (((d * e) + (a - b)) * ((c + d) / (e - a))));
        int megaExpr2 = (((((a * b) + (c * d)) - ((e * a) + (b * c))) / (((d + e) * (a + b)) - ((c * d) + (e * a)))));
        int megaExpr3 = ((((((a + b + c) * (d + e)) - ((a * b) + (c * d))) / (((e + a) * (b + c)) + ((d + e) * (a + b)))) + (((a - b) * (c - d)) + ((e - a) * (b - c)))));
        
        // Deep expression (safe bounds)
        int deepExpr = (((((a + b) * c) - d) / e) + a) * b;
        
        // Mathematical intensity
        int formula1 = (a * b * c) + (d * e * a) - (b * c * d) + (e * a * b) - (c * d * e);
        int formula2 = ((a + b + c + d + e) * (a - b - c - d - e)) / ((a * b) + (c * d) + (e * a));
        int formula3 = (((a / b) + (c / d)) * ((e / a) + (b / c))) - (((d / e) + (a / b)) * ((c / d) + (e / a)));
        
        // Computation chains
        int chainResult = 0;
        int tempVal = 1;
        tempVal = tempVal + a;
        chainResult = chainResult + tempVal;
        tempVal = tempVal * b;
        chainResult = chainResult + tempVal;
        tempVal = tempVal - c;
        chainResult = chainResult + tempVal;
        tempVal = tempVal / d;
        chainResult = chainResult + tempVal;
        tempVal = tempVal + e;
        chainResult = chainResult + tempVal;
        
        // Large scale summation
        int sumTotal = a + b + c + d + e + megaExpr1 + megaExpr2 + megaExpr3 + deepExpr + formula1 + formula2 + formula3 + chainResult;
        
        System.out.println("Extreme Arithmetic Results:");
        System.out.println("Mega expression 1:" + megaExpr1);
        System.out.println("Mega expression 2:" + megaExpr2);
        System.out.println("Deep nested result:" + deepExpr);
        System.out.println("Sum total:" + sumTotal);
        
        // === PHASE 2: MASSIVE VARIABLE OPERATIONS ===
        System.out.println("Phase 2: Extreme Variable Manipulation");
        
        // Create many variables and manipulate them
        int[] vars = new int[50];
        for (int i = 0; i < 50; i++) {
            vars[i] = i * 2 + 1;
        }
        
        // Complex variable interactions
        int varSum = 0;
        for (int i = 0; i < 50; i++) {
            vars[i] = vars[i] * (i + 1) / (i + 2);
            varSum = varSum + vars[i];
        }
        
        // Nested variable assignments
        int x1 = varSum / 10;
        int x2 = x1 * x1;
        int x3 = x2 - x1;
        int x4 = x3 + x2;
        int x5 = x4 / x3;
        
        System.out.println("Variable Results:");
        System.out.println("Variable sum:" + varSum);
        System.out.println("X1:" + x1 + " X2:" + x2 + " X3:" + x3 + " X4:" + x4 + " X5:" + x5);
        
        // === PHASE 3: EXTREME EXPRESSION NESTING ===
        System.out.println("Phase 3: Maximum Expression Complexity");
        
        // Ultra-deep nested expressions
        int ultra1 = ((((((a + b) * c) - d) / e) + ((((x1 + x2) * x3) - x4) / x5)) * ((((megaExpr1 + megaExpr2) * megaExpr3) - deepExpr) / formula1));
        int ultra2 = (((((((a * b) + (c * d)) - (e * x1)) + ((x2 * x3) - (x4 * x5))) * (((formula1 + formula2) - formula3) + chainResult)) / ((megaExpr1 + megaExpr2 + megaExpr3) - (deepExpr + sumTotal))) + varSum);
        int ultra3 = ((((x1 + x2 + x3 + x4 + x5) * (a + b + c + d + e)) - ((megaExpr1 + megaExpr2) * (formula1 + formula2))) / ((deepExpr + chainResult) + (varSum / 10)));
        
        // Extreme mathematical expressions
        int mathExtreme1 = 0;
        for (int i = 1; i <= 25; i++) {
            mathExtreme1 = mathExtreme1 + (i * i * i) / (i + 1) + (i * i) % 7;
        }
        
        int mathExtreme2 = 0;
        for (int i = 1; i <= 20; i++) {
            for (int j = 1; j <= 10; j++) {
                mathExtreme2 = mathExtreme2 + (i * j) % 13 + (i + j) / 3;
            }
        }
        
        System.out.println("Extreme Expression Results:");
        System.out.println("Ultra 1:" + ultra1);
        System.out.println("Ultra 2:" + ultra2);
        System.out.println("Ultra 3:" + ultra3);
        System.out.println("Math extreme 1:" + mathExtreme1);
        System.out.println("Math extreme 2:" + mathExtreme2);
        
        // === PHASE 4: COMPUTATIONAL OVERLOAD ===
        System.out.println("Phase 4: Maximum Computational Load");
        
        // Fibonacci stress test
        int fibStress = 0;
        for (int i = 1; i <= 12; i++) {
            fibStress = fibStress + fibonacci(i);
        }
        
        // Factorial combinations
        int factStress = 0;
        for (int i = 1; i <= 8; i++) {
            factStress = factStress + factorial(i) / (i + 1);
        }
        
        // Power calculations
        int powerStress = 0;
        for (int base = 2; base <= 5; base++) {
            for (int exp = 1; exp <= 8; exp++) {
                powerStress = powerStress + power(base, exp) % 1000;
            }
        }
        
        // Final extreme calculation
        int extremeResult = megaExpr1 + megaExpr2 + megaExpr3 + deepExpr + 
                           formula1 + formula2 + formula3 + chainResult + sumTotal +
                           varSum + ultra1 + ultra2 + ultra3 + 
                           mathExtreme1 + mathExtreme2 + fibStress + factStress + powerStress;
        
        System.out.println("Computational Overload Results:");
        System.out.println("Fibonacci stress:" + fibStress);
        System.out.println("Factorial stress:" + factStress);
        System.out.println("Power stress:" + powerStress);
        System.out.println("EXTREME FINAL RESULT:" + extremeResult);
        
        long endTime = System.nanoTime();
        long totalTimeMs = (endTime - startTime) / 1_000_000;
        
        System.out.println("Computation time:" + totalTimeMs);
        System.out.println("=== EXTREME BENCHMARK COMPLETE ===");
        System.out.println("Total execution time:" + totalTimeMs);
        System.out.println("Extreme phases completed: 4");
        System.out.println("Ultra-complex operations: 500+");
        System.out.println("Deep nested expressions: 50+");
        System.out.println("Computational stress tests: 20+");
        System.out.println("Final extreme score:" + totalTimeMs);
        System.out.println("=== Java Extreme Performance Test Complete ===");
    }
    
    // Helper functions
    static int fibonacci(int n) {
        if (n <= 1) return n;
        return fibonacci(n - 1) + fibonacci(n - 2);
    }
    
    static int factorial(int n) {
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }
    
    static int power(int base, int exp) {
        int result = 1;
        for (int i = 0; i < exp; i++) {
            result = result * base;
        }
        return result;
    }
}
